Add support for reading gstlal trigger files

dethodav opened the pull request:
This pull request aims to add support for reading gstlal trigger files.

The rationale for a specific gstlal format is 

* A desire to calculate some useful derived quantities. This feature is available for only pycbc trigger files at present. 
* The gstlal statistic requires additional inputs beyond what is included in trigger files in order to calculate the ranking statistic. However, the comparable quantity is stored in the `coinc_event` table. Hence reading in both the `coinc_event` and `coinc_inspiral` tables at once would be a useful feature. 

Thew new proposed format heavily relies upon the current ligolw functionality, with the addition of the two features above.

pep8speaks commented:
Hello @dethodav! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`gwpy/table/io/gstlal.py`](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py):

> [Line 40:1](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L40): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 50:45](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L50): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 69:80](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L69): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (81 > 79 characters)
> [Line 74:80](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L74): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)
> [Line 82:39](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L82): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 91:80](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L91): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (94 > 79 characters)
> [Line 93:13](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L93): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 93:23](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L93): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 93:25](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L93): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 93:80](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L93): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (88 > 79 characters)
> [Line 94:30](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L94): [E111](https://duckduckgo.com/?q=pep8%20E111) indentation is not a multiple of four
> [Line 94:30](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L94): [E113](https://duckduckgo.com/?q=pep8%20E113) unexpected indentation
> [Line 95:13](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L95): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 95:32](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L95): [E701](https://duckduckgo.com/?q=pep8%20E701) multiple statements on one line (colon)
> [Line 97:13](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L97): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 97:27](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L97): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 97:29](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L97): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 97:80](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L97): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (108 > 79 characters)
> [Line 98:13](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L98): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 98:24](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L98): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 98:26](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L98): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 98:80](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L98): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (99 > 79 characters)
> [Line 99:9](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L99): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 99:13](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L99): [E701](https://duckduckgo.com/?q=pep8%20E701) multiple statements on one line (colon)
> [Line 100:27](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L100): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 100:29](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L100): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 100:80](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L100): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (85 > 79 characters)
> [Line 101:24](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L101): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 101:26](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L101): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 102:9](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L102): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 102:15](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L102): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 102:17](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L102): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 104:9](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L104): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 104:33](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L104): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 104:35](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L104): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 105:5](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L105): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 105:9](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L105): [E701](https://duckduckgo.com/?q=pep8%20E701) multiple statements on one line (colon)
> [Line 107:5](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L107): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 107:31](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L107): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ':'
> [Line 107:31](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L107): [E701](https://duckduckgo.com/?q=pep8%20E701) multiple statements on one line (colon)
> [Line 107:32](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L107): [E701](https://duckduckgo.com/?q=pep8%20E701) multiple statements on one line (colon)
> [Line 108:15](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L108): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 108:17](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L108): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 109:5](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L109): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 112:1](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L112): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 112:1](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L112): [E266](https://duckduckgo.com/?q=pep8%20E266) too many leading '#' for block comment
> [Line 113:1](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L113): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 113:64](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L113): [E701](https://duckduckgo.com/?q=pep8%20E701) multiple statements on one line (colon)
> [Line 117:13](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L117): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 117:79](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L117): [E701](https://duckduckgo.com/?q=pep8%20E701) multiple statements on one line (colon)
> [Line 122:1](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L122): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 123:1](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L123): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 124:1](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L124): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 126:1](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L126): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 127:1](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L127): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 128:1](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L128): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 129:1](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L129): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 130:1](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L130): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 131:1](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L131): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 133:1](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L133): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 133:11](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L133): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 133:13](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L133): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 134:1](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L134): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 134:17](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L134): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 134:19](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L134): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 136:1](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L136): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 136:20](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L136): [E701](https://duckduckgo.com/?q=pep8%20E701) multiple statements on one line (colon)
> [Line 136:21](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L136): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 139:8](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L139): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 139:10](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L139): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 139:27](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L139): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ':'
> [Line 139:27](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L139): [E701](https://duckduckgo.com/?q=pep8%20E701) multiple statements on one line (colon)
> [Line 142:1](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L142): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 142:18](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L142): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 142:20](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L142): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 143:1](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L143): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 143:24](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L143): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 143:26](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L143): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 145:1](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L145): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 145:48](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L145): [E701](https://duckduckgo.com/?q=pep8%20E701) multiple statements on one line (colon)
> [Line 145:49](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L145): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 148:8](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L148): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 148:10](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L148): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 148:25](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L148): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ':'
> [Line 148:25](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L148): [E701](https://duckduckgo.com/?q=pep8%20E701) multiple statements on one line (colon)
> [Line 149:8](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L149): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 149:10](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L149): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 149:27](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L149): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ':'
> [Line 149:27](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L149): [E701](https://duckduckgo.com/?q=pep8%20E701) multiple statements on one line (colon)
> [Line 150:12](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L150): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 150:14](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L150): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 150:42](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L150): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 153:1](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L153): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 153:22](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L153): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 153:24](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L153): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 154:1](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L154): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 154:28](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L154): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 154:30](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L154): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 154:37](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L154): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 156:1](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L156): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 156:23](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L156): [E701](https://duckduckgo.com/?q=pep8%20E701) multiple statements on one line (colon)
> [Line 159:10](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L159): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 159:12](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L159): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 159:29](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L159): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ':'
> [Line 159:29](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L159): [E701](https://duckduckgo.com/?q=pep8%20E701) multiple statements on one line (colon)
> [Line 160:10](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L160): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 160:12](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L160): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 160:29](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L160): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ':'
> [Line 160:29](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L160): [E701](https://duckduckgo.com/?q=pep8%20E701) multiple statements on one line (colon)
> [Line 163:1](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L163): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 163:21](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L163): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 163:23](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L163): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 164:1](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L164): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 164:27](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L164): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 164:29](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L164): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 164:38](https://github.com/gwpy/gwpy/blob/5c9dbf3d0f50d367715068a54e572e5396399d79/gwpy/table/io/gstlal.py#L164): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','

dethodav commented:
At present, this pull request is not ready for merge - one of the remaining items is setting up the necessary registry in the format file. @duncanmmacleod @alurban Would you be able to help explain how the file registry is set up?

duncanmmacleod commented:
@dethodav, is this still a desired change? If 'yes' (I hope so) then I can help get things straightened out, tested, reviewed, and merged.

First up, please `rebase` against the upstream `master` branch.

dethodav commented:
@duncanmmacleod Yes, I would still be interested in adding this feature to gwpy. I've gone ahead and rebased to the current main branch and updated the PR. Let me know your suggested next steps.

duncanmmacleod commented:
@dethodav, I let this linger again, please rebase. I think the PEP8/flake linter issues are real, so please have a look at those.

Please think also about how to add tests of the new functionality.

duncanmmacleod commented on gwpy/table/io/gstlal.py:
```diff
@@ -0,0 +1,164 @@
+    return events
+
+
+### !!! unclear how to set up register
+def identify_gstlal(origin, filepath, fileobj, *args, **kwargs):
+    """Identify a PyCBC Live file as an HDF5 with the correct name
+    """
+    if identify_hdf5(origin, filepath, fileobj, *args, **kwargs) and (
+            filepath is not None and PYCBC_FILENAME.match(basename(filepath))):
+        return True
+    return False
```
@dethodav, the `identify_gstlal` function should take in the arguments are you have them listed, and inspect them to unimbiguously detect whether the incoming file is a 'gstlal file' (returning `True`) or not (`False`).

For PyCBC we do this with a regular expression on the filename, which isn't great, but seems to work well enough.

What is unique about a GstLAL file?

dethodav commented:
@duncanmmacleod Here is an example gstlal xml file (available at CIT): `/home/gstlalcbc/observing/4/dev/runs/trigs.edward_O3_replay_MDC06/13443/H1L1V1-0598_noninj_LLOID-1344312382-14414.xml.gz`

dethodav commented:
@duncanmmacleod I've gone through and made updates to this PR. Things should be ready for review modulo the failing "Dependencies" check. Do you have any suggestions on how best to resolve this import issue?

duncanmmacleod reviewed (changes requested):
> @duncanmmacleod I've gone through and made updates to this PR. Things should be ready for review modulo the failing "Dependencies" check. Do you have any suggestions on how best to resolve this import issue?

Thanks @dethodav, the fundamental issue here is that `python-ligo-lw` is not a first-class requirement of GWpy, it's optional. So you need to protect the imports in `gwpy/table/io/gstlal.py` and `gwpy/table/tests/test_io_gstlal.py` so that the test suite executes without it.

Refer to `gwpy/table/io/ligolw.py` and `gwpy/table/tests/test_io_ligolw.py` for inspiration there.

You will also need to add a reference to the new `gstlal` module in `gwpy/table/io/__init__.py` so that the I/O format is registered automatically when the `EventTable` is first imported.

dethodav commented:
@duncanmmacleod I've now addressed the rest-related issues. Let me know if there's anything else that needs to be fixed before merger.

duncanmmacleod reviewed (approved):
Thanks for this @dethodav, looks great.
